luigi/s3.py

```diff
@@ -449,7 +456,7 @@ def move(self, source_path, destination_path, **kwargs):
         self.copy(source_path, destination_path, **kwargs)
         self.remove(source_path)
 
-    def listdir(self, path, start_time=None, end_time=None):
+    def listdir(self, path, start_time=None, end_time=None, with_key=False):
         """
         Get an iterable with S3 folder contents.
         Iterable contains paths relative to queried path.
```

Comment:
Maybe also add docs like `:param with_key: if True this function will instead return blah blah`?